File: LeetCode/CPP/LC-13.cpp
class Solution {
public:
    int romanToInt(string s) {
        int str_length = s.length(), int_number = 0;
        
        for(int index = 0; index < str_length; index++)
        {
            if(s[index] == 'M') {
                int_number += 1000;
            }
            else if(s[index] == 'D') {
                int_number += 500;
            }
            else if(s[index] == 'C') {
                if(index != (str_length - 1) && (s[index+1] == 'D' or s[index+1] == 'M')) {
                int_number -= 100;
            }
                else {
                    int_number += 100;
                }
            }
            else if(s[index] == 'L') {
                int_number += 50;
            }
            else if(s[index] == 'X') {
                if (index != (str_length-1) and (s[index+1] == 'L' or s[index+1] == 'C')) {
                    int_number -= 10;
                }
                else {
                    int_number += 10;
                }
            }
            else if(s[index] == 'V') {
                int_number += 5;
            }
            else if(s[index] == 'I') {
                if(index != (str_length-1) and (s[index+1] == 'V' or s[index+1] == 'X')) {
                    int_number -= 1;
                }
                else {
                    int_number += 1;
                }
            }
            
        }
        return int_number;
    }
};

File: LeetCode/Python/LC-13.py
class Solution:
    def romanToInt(self, s: str) -> int:
        integer = 0
        length_s = len(s)
        i = 0

        while i < length_s:
            if s[i] == 'I':
                if i != length_s - 1:
                    if s[i+1] == 'V':
                        integer += 4
                        i += 2
                        continue
                    if s[i+1] == 'X':
                        integer += 9
                        i += 2
                        continue

                integer += 1

            if s[i] == 'X':
                if i != length_s - 1:
                    if s[i+1] == 'L':
                        integer += 40
                        i += 2
                        continue
                    if s[i+1] == 'C':
                        integer += 90
                        i += 2
                        continue

                integer += 10

            if s[i] == 'C':
                if i != length_s - 1:
                    if s[i+1] == 'D':
                        integer += 400
                        i += 2
                        continue
                    if s[i+1] == 'M':
                        integer += 900
                        i += 2
                        continue

                integer += 100

            if s[i] == 'V':
                integer += 5

            if s[i] == 'L':
                integer += 50

            if s[i] == 'D':
                integer += 500

            if s[i] == 'M':
                integer += 1000
            
            i +=1

        return integer